### PR TITLE
refactor: Remove dependency on `pipelinewise-singer-python` and move singer library code into `singer_sdk._singerlib`

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Related issues and discussions:

- Closes https://github.com/meltano/sdk/issues/928
- Blocks https://github.com/meltano/sdk/issues/977
- Relates to https://github.com/meltano/sdk/issues/208 but we won't commit to interop or parity with existing libraries for the time being.

----
:books: Documentation preview :books:: https://meltano-sdk--979.org.readthedocs.build/en/979/

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -52,6 +51,9 @@ inflection = "^0.5.1"
 sqlalchemy = "^1.4"
 python-dotenv = ">=0.20,<0.22"
 typing-extensions = "^4.2.0"
+simplejson = "^3.17.6"
+jsonschema = "^4.16.0"
+pytz = "^2022.2.1"
```
Comment (edgarrmondragon) [inline pyproject.toml:56]: These were dependencies of `pipelinewise-singer-python`, so I'm adding them explicitly here.

File: pyproject.toml [hunk 2]
```diff
@@ -149,7 +153,7 @@ exclude_lines = [
     "if __name__ == .__main__.:",
     '''class .*\bProtocol\):''',
     '''@(abc\.)?abstractmethod''',
-    "if TYPE_CHECKING:",
+    '''if (t\.)?TYPE_CHECKING:''',
```
Comment (edgarrmondragon) [inline pyproject.toml:156]: So we can `import typing as t`, which is sometimes convenient.

File: tests/_singerlib/test_schema.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import pytest
```
Comment (edgarrmondragon) [inline tests/_singerlib/test_schema.py:1]: The tests in this module are adapted from https://github.com/transferwise/pipelinewise-singer-python/blob/master/tests/test_schema.py.

File: tests/_singerlib/test_catalog.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
+from typing import List, Optional
```
Comment (edgarrmondragon) [inline tests/_singerlib/test_catalog.py:1]: Added some more tests here for catalog and metadata manipulation to ensure 100% coverage in the `singer_sdk._singerlib` module.

**Comment by edgarrmondragon** [comment]
@aaronsteers @kgpayne this is ready for review 😄

**Review by aaronsteers — APPROVED** [review]
🚀

**Pull request merged**